### PR TITLE
Added batch normalization & return training history

### PR DESCRIPTION
**Pull request by dhing1024**

### CODE REVIEW

**Review by CaptainSpaceCat — COMMENTED** [review]
glad the batchnorm is working
